General Description
	For one of my dissertation chapters I will be working with a population of diverse Plant Introductions(PIs) sampled from the 
	USDA Germplasm resources Information Network (GRIN). The plan is to do genome-wide association studies to identify 
	quantitative disease resistant loci using the population. The GWAS will be developed in greenhouse and field environments. Due
	to complexities of field studies for quantitative disease resistance to Phytopthora sojae I will be resticted to the number 
	of lines in my study. Ideally GWAS would be developed using a large diverse population. Therefore, in an attempt to maximize 
	GWAS potential with minimal population sizes and to fulfil the class requirements, I propose to do a population structure 
	analysis of the PIs with a 50K SNP BeadChip(soybean). Furthermore I plan to analyze population structure in regions found to 
	be highly associated with quantitative resistance in prior studies through haplotype visualization.
Technical aspects
	While there are packages available in R to do the population analysis I will be attempting to complete this in python. I will
	need scripts to remove monomorphic markers, markers with > 5% missing data, and markers with a minor allele frequency of < 5% 
	prior to analysis. I will need a script for the population structure analysis and one for the haplotype visualization. I plan
	to compare fastSturucture and Stucture, both written in python. I am not sure how I will be running the haplotype visualization
	yet but have found scripts for human studies that I might use as a referance. 
Project uncertainty 
	I am mainly uncertain of how I will be running the haplotype analysis. 
Reasoning
	I choose this as my final project because it will be useful to my research since I will be able to gain knowledge on the 
	diversity of my population. Additionally, it will allow me to practice the topics covered in the course by implementing them
	to my own studies. 
